include/oneapi/tbb/concurrent_queue.h

```diff
@@ -137,25 +137,29 @@ class concurrent_queue {
     }
 
     concurrent_queue& operator=( const concurrent_queue& other ) {
-        //TODO: implement support for std::allocator_traits::propagate_on_container_copy_assignment
-        if (my_queue_representation != other.my_queue_representation) {
-            clear();
+        if (my_queue_representation == other.my_queue_representation)
+            return *this;
+        clear();
+        if (queue_allocator_traits::propagate_on_container_move_assignment::value) {
             my_allocator = other.my_allocator;
-            my_queue_representation->assign(*other.my_queue_representation, my_allocator, copy_construct_item);
         }
+        my_queue_representation->assign(*other.my_queue_representation, my_allocator, copy_construct_item);
         return *this;
```

Comment:
I propose to rewrite consistent with other containers and use TBB helper function for copy assigning the allocators (`tbb::detail::copy_assign_allocators)`:

```
if (this != &other) {
    clear();
    copy_assign_allocators(my_allocator, other.my_allocator);
    my_queue_representation->assign(*other.my_queue_representation, my_allocator, copy_construct_item);
}
return *this;
```